### PR TITLE
UPDATE 76

### PR DESCRIPTION
**Pull request by Logandros**
* Added HK-55 Mode for that pesky 2 hours of having to play him
* Assassin / Deception AoE was totally broken
* Guardian / Vigilance movement was missing from melee argument, dps rotation needed a total re-write (thanks Zemnexx)
* Guardian / Defense optimized (thanks Zemnexx)

### CODE REVIEW

**Comment by Logandros** [comment]
UPDATE 76

* Added HK-55 Mode for that pesky 2 hours of having to play him
* Assassin / Deception AoE was totally broken
* Guardian / Vigilance movement was missing from melee argument, dps rotation needed a total re-write (thanks Zemnexx)
* Guardian / Defense optimized (thanks Zemnexx)

File: trunk/DefaultCombat/Helpers/CombatHotkeys.cs [hunk 1]
```diff
@@ -114,5 +119,19 @@ private static void ChangeSolo()
 				EnableSolo = true;
 			}
 		}
+
+		private static void ChangeHK55()
```
Comment (aevitas) [inline trunk/DefaultCombat/Helpers/CombatHotkeys.cs:123]: You can write this more succinctly:

```
private static void ToggleHk55()
{
    EnableHK55 = !EnableHK55;
    Logger.Write("HK-55 mode enabled: " + EnableHK55);
}
```

File: trunk/DefaultCombat/Routines/Advanced/Guardian/Defense.cs [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 // See the file LICENSE for the source code's detailed license
 
+using System.Windows.Input;
 using Buddy.BehaviorTree;
+using Buddy.CommonBot;
+using Buddy.Swtor;
+using Buddy.Swtor.Objects;
 using DefaultCombat.Core;
 using DefaultCombat.Helpers;
+using Targeting = DefaultCombat.Core.Targeting;
```
Comment (aevitas) [inline trunk/DefaultCombat/Routines/Advanced/Guardian/Defense.cs:11]: What is this conflicting with? Afaik there aren't multiple `Targeting` types, is it clashing between routine and bot core?

File: trunk/DefaultCombat/Routines/Advanced/Guardian/Defense.cs [hunk 3]
```diff
@@ -48,8 +53,9 @@ public override Composite SingleTarget
 			get
 			{
 				return new PrioritySelector(
-					Spell.Cast("Saber Throw", ret => !DefaultCombat.MovementDisabled && Me.CurrentTarget.Distance >= 1f && Me.CurrentTarget.Distance <= 3f),
-					Spell.Cast("Force Leap", ret => CombatHotkeys.EnableCharge && Me.CurrentTarget.Distance >= 1f && Me.CurrentTarget.Distance <= 3f),
+					Spell.Cast("Dispatch", ret => Me.HasBuff("Ardent Advocate")),
+                    Spell.Cast("Saber Throw", ret => Me.CurrentTarget.Distance >= 1f && Me.CurrentTarget.Distance <= 3f),
```
Comment (aevitas) [inline trunk/DefaultCombat/Routines/Advanced/Guardian/Defense.cs:57]: Nitpicky - but spacing.

**Review by aevitas — CHANGES_REQUESTED** [review]
Couple of minor things - feel free to implement or ignore as you see fit.

**Review by aevitas — APPROVED** [review]

**Pull request merged**